### ISSUE TITLE
Invalid Tag message for multiple helm charts

### ISSUE THREAD

**Issue by nicl-dev** [bitnami/charts#31044]
### Name and Version

bitnami/external-dns-8.7.0, bitnami/keycloak-24.3.0, bitnami/thanos-15.9.2

### What architecture are you using?

amd64

### What steps will reproduce the bug?

After trying to upgrade our helm-charts we cannot pull newer chart versions. For example external-dns:

```
helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "bitnami" chart repository
Update Complete. ⎈Happy Helming!⎈

helm pull --untar --repo https://charts.bitnami.com/bitnami external-dns --version 8.7.0
Error: invalid_reference: invalid tag
```

However, older versions like `external-dns 7.2.0` do work. The same thing happens with our keycloak and thanos charts. The versions are all taken from artifacthub.io.

### What is the expected behavior?

Helm pulling the new chart version.

### What do you see instead?

`Error: invalid_reference: invalid tag`

**Comment by nicl-dev** [comment]
And right after typing this it works, weird. Sorry for the confusion.

**Comment by Jasper-Ben** [comment]
Nah, it's not just you. We have the same issue, e.g. upgrading from redis 20.3.0 to anything newer.

**Comment by Jasper-Ben** [comment]
Between redis 20.3.0 (the last version we could successfully upgrade to) and redis 20.4.0 there have been changes to github workflows that seem to be relevant for tagging. My uneducated guess is that there is a bug somewhere there. In the redis chart there haven't been any relevant changes that could cause this issue.

**Comment by carrodher** [comment]
As reported in https://github.com/helm/helm/issues/13466, this problem stems from how the `helm ... --repo ...` command handles an _index.yaml_ file containing OCI URLs.

### Why is this issue appearing now?
Historically, Bitnami Helm charts were accessed by adding the Bitnami repository to the Helm CLI:
```
$ helm repo add bitnami https://charts.bitnami.com/bitnami
```
or by directly installing/pulling a specific Helm chart using the `--repo` flag:
```
$ helm pull --repo https://charts.bitnami.com/bitnami foo
```

In April 2023, [Bitnami introduced support for the OCI format](https://blog.bitnami.com/2023/04/httpsblog.bitnami.com202304bitnami-helm-charts-now-oci.html), recommending OCI-based installation as the primary method going forward. For example:
```
helm pull oci://registry-1.docker.io/bitnamicharts/external-dns --version 8.6.0
```

As detailed in our [recent announcement](https://blog.bitnami.com/2024/10/bitnami-helm-charts-moving-to-oci.html) and https://github.com/bitnami/charts/issues/30110, the traditional method is now being deprecated. Consequently, the _index.yaml_ served at `https://charts.bitnami.com/bitnami` now uses OCI URLs, exposing the above mentioned issue in the Helm CLI's handling of these URLs.

**Comment by nicl-dev** [comment]
Well, another case of RTFM. I was looking for the cause in your most recent announcement about Bitnami Premium because I had a feeling something like this might have happened, but unfortunately, I didn’t scroll far enough. Thank you for pointing me in the right direction.

**Comment by nicl-dev** [comment]
Well, another case of RTFM. I was looking for the cause in your most recent announcement about Bitnami Premium because I had a feeling something like this might have happened, but unfortunately, I didn’t scroll far enough. Thank you for pointing me in the right direction.
